clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -86,6 +87,19 @@ class AzureResourceManagerClient extends AzureBaseClient {
     }
   }
 
+  ResourceGroup initializeResourceGroupAndVNet(AzureCredentials creds, String resourceGroupName, String region) {
+    ResourceGroup resourceGroup
+    if (!resourceGroupExists(creds, resourceGroupName)) {
```

Comment:
Can you call the get for a resourcegroup and if it fails then do the create rather than doing a check and then possibly a get? If a get results in an exception rather than null if it doesn't exist then the approach here would be better. Not a big deal either way though so feel free to ignore.